macros.cfg

```diff
@@ -148,24 +148,53 @@ description: Prints a primeline, used internally, if configured, as part of the
 gcode:
   SAVE_GCODE_STATE NAME=prime_line_state
   {% set speed = printer["gcode_macro RatOS"].macro_travel_speed|float * 60 %}
+  {% set prime_x0 = params.X0|default(5)|float %}
+  {% set prime_y0 = params.Y0|default(5)|float %}
+  {% set prime_len = params.LENGTH|default(70)|float %}
+  {% set prime_wipe_len = params.WIPE_LENGTH|default(20)|float %}
+  {% set prime_axis = params.AXIS|default("Y") %}
+  {% if prime_axis not in "XY" %}
+    {% set prime_axis = "Y" %}
+  {% endif %}
+  {% if params.AXIS_INVERT %}
+    {% set prime_len = -prime_len %}
+    {% set prime_wipe_len = -prime_wipe_len %}
+  {% endif %}
+  # for some reason, the nozzle consistently goes cold if the entire line is
+  # ordered in a single G1, but holds true if the line is broken up into
+  # pieces; it's almost like the PID loop only runs between G1s
+  {% set prime_segments = params.SEGMENTS|default(8)|int %}
+  {% set filament_cross_section_area = 2.405281875 %} # approximately pi*1.75*1.75/4
+  {% set nozzle_diameter = params.NOZZLE_DIAMETER|default(0.4)|float %}
+  {% set prime_layer_height = nozzle_diameter * 5 / 8 %}
+  # nozzles of varying bores seem to have flats of constant wall-thickness;
+  # so if we used a percentage of nozzle-bore, then we'd be more likely to
+  # extrude past the flats, the bigger the nozzle we were being asked about.
+  # so it's unfortunate that SuperSlicer's width settings can't do
+  # what we're about to do here:
```

Comment:
Given the perennial problem of link rot, I don't know how to cram a photograph like this into a comment in the code.

Thus, I'll just put here an example that clearly shows the nozzle-flats at various bores of one popular line of nozzles, the Bondtech CHT in nickel-coated brass.

![image](https://user-images.githubusercontent.com/992859/165876775-15adb2de-bc4d-42c8-b022-9d67a9dc6cdc.png)